weather_app/src/Components/Weather.jsx

```diff
@@ -1,50 +1,63 @@
-import React, {useState} from 'react';
-import axios from 'axios'
+import React, { useState } from "react";
+import axios from "axios";
 
 const Weather = () => {
-    const [data, setData] = useState({})
-    const [location, setLocation] = useState('')
+  const [data, setData] = useState({});
+  const [location, setLocation] = useState("");
 
-    const searchLocation = (event) => {
-        if (event.key === 'Enter') {
-            axios.get(url).then((response) => {
-                setData(response.data)
-                console.log(response.data)
-            })
-            setLocation("")
-        }
+  const searchLocation = (event) => {
+    if (event.key === "Enter") {
+      // all async request should be handled with trycatch
+      axios.get(url).then((response) => {
+        setData(response.data);
+        console.log(response.data);
+      });
+      setLocation("");
     }
+  };
+  // this var belongs to searchLocation function only, not for all component
```

Comment:
this var belongs to searchLocation function only, not for all component
React create this variable every render even if threris no call of searchLocation

Suggested to include this var to the function